### ISSUE TITLE
Unable to install A11y Machine due to phantomjs error

### ISSUE THREAD

**Issue by rakeshkmr516** [liip/TheA11yMachine#110]
Hi ,
I am unable to install sudo npm install -g the-a11y-machine due tophantom js error 
since phantomjs days came to an end, can we remove this package from the bundle 

> phantomjs-prebuilt@2.1.15 install /usr/local/lib/node_modules/the-a11y-machine/node_modules/phantomjs-prebuilt
> node install.js

Considering PhantomJS found at /usr/local/bin/phantomjs
Looks like an `npm install -g`
Found PhantomJS at /usr/local/lib/node_modules/phantomjs/lib/phantom/bin/phantomjs ...verifying
Writing location.js file
Error checking path, continuing { Error: EACCES: permission denied, open '/usr/local/lib/node_modules/the-a11y-machine/node_modules/phantomjs-prebuilt/lib/location.js'
    at Object.fs.openSync (fs.js:652:18)
    at Object.fs.writeFileSync (fs.js:1299:33)
    at writeLocationFile (/usr/local/lib/node_modules/the-a11y-machine/node_modules/phantomjs-prebuilt/lib/util.js:84:6)
    at Promise._successFn (/usr/local/lib/node_modules/the-a11y-machine/node_modules/phantomjs-prebuilt/install.js:366:11)
    at nextTickCallback (/usr/local/lib/node_modules/the-a11y-machine/node_modules/kew/kew.js:47:28)
    at _combinedTickCallback (internal/process/next_tick.js:131:7)
    at process._tickCallback (internal/process/next_tick.js:180:9)
  errno: -13,
  code: 'EACCES',
  syscall: 'open',
  path: '/usr/local/lib/node_modules/the-a11y-machine/node_modules/phantomjs-prebuilt/lib/location.js' }
Download already available at /tmp/phantomjs/phantomjs-2.1.1-macosx.zip
Verified checksum of previously downloaded file
Extracting zip contents
Removing /usr/local/lib/node_modules/the-a11y-machine/node_modules/phantomjs-prebuilt/lib/phantom
Copying extracted folder /tmp/phantomjs/phantomjs-2.1.1-macosx.zip-extract-1506325827577/phantomjs-2.1.1-macosx -> /usr/local/lib/node_modules/the-a11y-machine/node_modules/phantomjs-prebuilt/lib/phantom
Phantom installation failed { Error: EACCES: permission denied, link '/tmp/phantomjs/phantomjs-2.1.1-macosx.zip-extract-1506325827577/phantomjs-2.1.1-macosx' -> '/usr/local/lib/node_modules/the-a11y-machine/node_modules/phantomjs-prebuilt/lib/phantom'
  errno: -13,
  code: 'EACCES',
  syscall: 'link',
  path: '/tmp/phantomjs/phantomjs-2.1.1-macosx.zip-extract-1506325827577/phantomjs-2.1.1-macosx',
  dest: '/usr/local/lib/node_modules/the-a11y-machine/node_modules/phantomjs-prebuilt/lib/phantom' } Error: EACCES: permission denied, link '/tmp/phantomjs/phantomjs-2.1.1-macosx.zip-extract-1506325827577/phantomjs-2.1.1-macosx' -> '/usr/local/lib/node_modules/the-a11y-machine/node_modules/phantomjs-prebuilt/lib/phantom'

**Comment by Hywan** [comment]
I am not able to reproduce your issue.

> since phantomjs days came to an end, can we remove this package from the bundle

I don't understand what your are suggesting. `a11ym` uses PhantomJS to test the web pages. We would like to use [SlimerJS](https://slimerjs.org/) one day, and probably [libservo](https://groups.google.com/forum/#!msg/mozilla.dev.servo/20lkEsRI-ZI/RbEaVG_MAAAJ) one other day too, but it's the case today. If you have time to help on this area, we would be glad to work with you!

**Comment by rakeshkmr516** [comment]
Finally able to install with yarn  Thank you @Hywan  for your support 
I am able to run on an example website which gave me good report with issues ( Accessibility errors) under a11ym_output index file. Kudos to your team

However when i tried on my real website which requires authentication able to fetch urls until redirection failed to run accessibility test on URLS

**Comment by Hywan** [comment]
`--http-auth-*` options are dedicated to solve this issue. What is SAML login page?

**Comment by rakeshkmr516** [comment]
we are using saml based authentication after crawling to that particular url fetch stopped and run havent executed

**Comment by rakeshkmr516** [comment]
@Hywan 
I am able to successfully point it at sites that do not require login. However, our application does require login and the “–http-auth-user” and “–http-auth-password” parameters appear to do nothing. Here is my sample usage. I would guess I am doing something wrong.

a11ym –http-auth-user username –http-auth-password password -http-tls-disable https://website.com

Are these headers confirmed to work? I could not find much on how to use them properly.

Thank you for your help, and have a wonderful day

**Comment by Hywan** [comment]
HTTP auth works yes, but it's basic auth, not SAML.

SAML support is not implemented yet.

**Comment by rakeshkmr516** [comment]
@Hywan  Great to hear Could you please take it as an enhancement since there are so many form based authentication apps

**Comment by Hywan** [comment]
My recommendation is to have a “test-xxx” profile for your application that by-pass some auth for instance. I am no longer working actively on this project since I don't work for Liip anymore. I am just a regular contributor.

cc @jeanmonod @ralf57
